### PR TITLE
Refactors tile reskinning

### PR DESCRIPTION
**Pull request by Rohesie**
## About The Pull Request

Refactored the tile reskinning and associated code to offer more generic and expansible behavior.

* Variant floors now produce the proper variant item stack when crowbared.
* Removes `icon_regular_floor` and associated old code, replacing it with a more standard use of `base_icon_state`.
* Renames the icon states to something more informative.
* Replaces the `human_maxHealth` var for a define.

## Why It's Good For The Game

@KathrinBailey wants to expand on this feature, so to avoid repeated code a more generic approach is positive.

dreamseeker_ZVCgi7UxrP

### CODE REVIEW

File: code/modules/mob/living/living_defines.dm [hunk 1]
```diff
@@ -12,8 +12,10 @@
 	var/lastattackerckey = null
 
 	//Health and life related vars
-	var/maxHealth = 100 ///Maximum health that should be possible.
-	var/health = 100 	///A mob's health
+	/// Maximum health that should be possible.
+	var/maxHealth = MAX_LIVING_HEALTH
+	/// The mob's current health.
+	var/health = MAX_LIVING_HEALTH
```
Comment (ShizCalev) [inline code/modules/mob/living/living_defines.dm:18]: this should be in a separate PR.

**Review by ShizCalev — DISMISSED** [review]

File: code/modules/mob/living/living_defines.dm (hunk 1 above)
Comment (Rohesie) [reply code/modules/mob/living/living_defines.dm:18]: It will conflict with the removal of `human_maxHealth`, but sure.

**Review by Twaticus — APPROVED** [review]

**Comment by KathrinBailey** [comment]
There's an issue I found that I poked Rohesie in DMs.
image
All of these become different type paths, so they cannot stack.
Because we change the name, desc and icon_state, all of these stacks appear identical to the player; they'll probably think this is a bug.
image
Here's me examining the different stacks, showing the player perspective.
image

File: code/__DEFINES/mobs.dm [hunk 2]
```diff
@@ -67,9 +67,7 @@
 #define LARVA_BODYPART "larva"
 /*see __DEFINES/inventory.dm for bodypart bitflag defines*/
 
-// Health/damage defines
-#define MAX_LIVING_HEALTH 100
```
Comment (Cyberboss) [inline code/__DEFINES/mobs.dm:71]: Why remove this?

Comment (Rohesie) [reply code/__DEFINES/mobs.dm:71]: This wasn't intentional, I actually atomized this change and it seems git auto-resolve of conflicts pulled a prank on me.
Fixing it.
